src/Microsoft.AspNet.SignalR.Client/Transports/WebSocketTransport.cs

```diff
@@ -166,13 +175,14 @@ internal virtual void OnClose()
         // fire and forget
         private async void DoReconnect()
         {
+            var token = CreateLinkedCancellationToken();
             var reconnectUrl = UrlBuilder.BuildReconnect(_connection, Name, _connectionData);
 
             while (TransportHelper.VerifyLastActive(_connection) && _connection.EnsureReconnecting())
             {
                 try
                 {
-                    await PerformConnect(reconnectUrl);
+                    await PerformConnect(reconnectUrl, token);
```

Comment:
My one concern is that we're using the same `_webSocketTokenSource` in this loop whereas before it would be reset everytime. Do you know if that is ok? If `LostConnection` is called does this loop matter anymore?

Reply:
You're probably right that it's better to move `var token = CreateLinkedCancellationToken();` inside the loop to avoid a possible regression.

The only non-terminal cancellation of _webSocketTokenSource is in LostConnection() which I don't think can happen until after the connection is reconnected, so I was trying to fix a possible race with dispose here by moving the creation of the _webSocketTokenSource outside the loop, but that's not the bug I was originally trying to fix with this PR.